cpp/src/arrow/util/bit_block_counter.h

```diff
@@ -46,7 +46,7 @@ inline uint64_t ShiftWord(uint64_t current, uint64_t next, int64_t shift) {
 }
 
 // These templates are here to help with unit tests
-
+// Two Arguments
 template <typename T>
 constexpr T BitNot(T x) {
```

Comment:
This comment only applies to `BitBlockAnd` and `BitBlockAndNot`. IMO, I would remove these comments because the function's signature if clear enough to identify the number of arguments.

Reply:
Ok Done.